pkg/restate/observability/categorize.go

```
@@ -0,0 +1,86 @@
+	if !ok {
+		return OutcomeFailed, CategoryInfra
+	}
+
+	code, parseErr := codes.ParseURN(urn)
+	if parseErr != nil {
+		return OutcomeFailed, CategoryInfra
+	}
+
+	// Workflow categories take precedence — they are explicitly set by call
+	// sites to convey blame regardless of the underlying System.
+	switch code.Category {
```

golangci-lint (2.11.4)

[error] missing cases in switch of type codes.Category: codes.CategoryUserBadRequest, codes.CategoryUserUnprocessableEntity, codes.CategoryUserTooManyRequests, codes.CategoryNotFound, codes.CategoryBadGateway, codes.CategoryServiceUnavailable, codes.CategoryGatewayTimeout, codes.CategoryUnauthorized, codes.CategoryForbidden, codes.CategoryRateLimited, codes.CategoryInternalServerError, codes.CategoryUnkeyData, codes.CategoryUnkeyAuthentication, codes.CategoryUnkeyAuthorization, codes.CategoryUnkeyLimits, codes.CategoryUnkeyApplication, codes.CategoryUnkeyVault

(exhaustive)